Add papers of week 19 and week 20

zhhangBian opened the pull request:
Add papers about cachedAttention and parallel series

# Contributing to LEMONADE reviews

Please review the project guidelines and check the boxes below.

**Subsections:**

- [x] Place papers only in **Level 3** or **Level 4** subsections.
- [x] Each subsection state the **common challenge**.
- [x] Each subsection has **at least 2 papers**.
- [x] Each subsection has **no more than 5 papers**.

**Tags:**

- [x] Tags are **specific techniques or contributions**
- [x] **No** broad area tags (like "performance") were used.
- [x] **No** vague feature tags (like "efficiency") were used *alone*.
- [x] Any potentially unclear acronyms in tags are **explained**.

youweizhuo commented:
Good. Please refer to my comment in #9 and #11. You can append motivation/challenge to the subsection summary. For example, after reading this pr, one might wonder: why data/model parallelism?

zhhangBian commented:
Here are the updates I've made:

- Enhanced the description for the 'common challenge' section concerning MP and TP.
- Added the papers for Week 20, specifically:
  - Distributed Parallelism: Included papers focusing on DP and MapReduce.
  - Heterogeneous Systems: Added papers discussing offloading tensors between CPU and GPU.
  - LLM Safety: Included the ICLR best paper on this topic.

Please review these changes.
